org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/util/nodeselection/NodeSelection.java

```diff
@@ -135,7 +136,7 @@ public boolean allMatch(Predicate<Object> predicate) {
      * @param newSelection the node selection to merge with
      * @return a new instance containing the merge result
      */
-    public NodeSelection mergeWith(NodeSelection newSelection) {
+    public NodeSelection mergeWith(IStructuredSelection newSelection) {
```

Comment:
Do we have to prepare to merge with selections other than NodeSelection? I think the argument should be as it was before.